src/components/Layer/Layer.js

```diff
@@ -1,38 +1,101 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-
-import core from 'core';
 
+import Icon from 'components/Icon';
 import Input from 'components/Input';
 
+import core from 'core';
+
 import './Layer.scss';
 
 class Layer extends React.PureComponent {
   static propTypes = {
     layer: PropTypes.object.isRequired,
-    layers: PropTypes.arrayOf(PropTypes.object).isRequired,
-    index: PropTypes.number.isRequired,
-  }
+  };
 
+  state = {
+    isExpanded: false,
+  };
+
+  unCheckChildren = (layer) => {
+    // new references for redux state
+    const newLayer = {...layer};
+    layer.children && layer.children.forEach((childLayer, i) => {
+      let newChildLayer = {...childLayer};
+      newChildLayer.visible = false;
+      newChildLayer = this.unCheckChildren(newChildLayer);
+      newLayer.children[i] = newChildLayer;
+    });
+    return newLayer;
+  }
 
   onChange = e => {
-    const { index, layers } = this.props;
+    const { updateLayer, layer } = this.props;
 
-    layers[index].visible = e.target.checked;
-    const doc = core.getDocument();
-    doc.setLayersArray(layers);
+    // new references for redux state
+    let newLayer = {...layer};
+    newLayer.visible = e.target.checked;
+    if (e.target.checked === false) {
+      newLayer = this.unCheckChildren(newLayer);
+    }
+    updateLayer(newLayer);
+  };
 
-    window.docViewer.refreshAll();
-    window.docViewer.updateView();
-  }
+  onClickExpand = () => {
+    this.setState(prevState => ({
+      isExpanded: !prevState.isExpanded,
+    }));
+  };
 
   render() {
-    const { layer } = this.props;
+    const { isExpanded } = this.state;
+    const { layer, updateLayer } = this.props;
+    const hasSubLayers = layer.children.length > 0;
 
     return (
       <div className="Layer">
-        <Input id={layer.name} type="checkbox" label={layer.name} onChange={this.onChange} defaultChecked={layer.visible} />
+        <div className="layer-wrapper">
+          <div className="padding">
+            {hasSubLayers && (
+              <div
+                className={`arrow ${
+                  isExpanded ? 'expanded' : 'collapsed'
+                }`}
+                onClick={this.onClickExpand}
+              >
+                <Icon glyph="ic_chevron_right_black_24px" />
+              </div>
+            )}
+          </div>
+          <Input
+            id={layer.name}
+            type="checkbox"
+            label={layer.name}
+            onChange={this.onChange}
+            checked={layer.visible}
+          />
+        </div>
+        {hasSubLayers && isExpanded && (
+          <div className="sub-layers">
+            {layer.children.map((subLayer, i) => (
+              <Layer
+                key={i}
+                index={i}
```

Comment:
Do we still need to pass `index` and `layers` down as props?

Reply:
Nope. Removed.